packages/web3-eth-ens/src/lib/ResolverMethodHandler.js

```diff
@@ -208,4 +230,31 @@ ResolverMethodHandler.prototype.prepareArguments = function (name, methodArgumen
     return [node];
 };
 
+/**
+ *
+ *
+ * @param {Contract} resolver
+ * @param {string} methodName
+ *
+ * @returns {Promise}
+ */
+ResolverMethodHandler.prototype.checkInterfaceSupport = async function (resolver, methodName) {
+    // Skip validation for undocumented interface ids (ex: multihash)
+    if (!interfaceIds[methodName]) return;
+
+    var supported = false;
+    try {
+        supported = await resolver
+            .methods
+            .supportsInterface(interfaceIds[methodName])
+            .call();
+    } catch(err) {
+        console.warn('Could not verify interface of resolver contract at "' + resolver.options.address + '". ');
```

Comment:
Couldn't it also throw other errors? ;-)

Reply:
Yes, it could but is spliced before another call to the resolver. If there's a problem with the connection or something, the error will be thrown then.